Application/components/Navbar/notifcation.tsx

```diff
@@ -85,20 +85,24 @@ const InboxDropdown: FC<InboxDropdownProps> = ({ userId }) => {
       if (!response.ok) {
         throw new Error('Failed to accept friend request');
       }
-      fetchFriendRequests();
+      fetchFriendRequests(); // Refresh data after accepting a request
     } catch (error) {
       console.error('Failed to accept friend request:', error);
     }
   };
 
   useEffect(() => {
-    if (userId) fetchFriendRequests();
+    if (userId) {
+      fetchFriendRequests(); // Initial fetch
+      const intervalId = setInterval(fetchFriendRequests, 15000); // Poll every 15 seconds
+      return () => clearInterval(intervalId); // Cleanup on component unmount
+    }
```

Comment:
Seems like 15 second is a good interval for an update on the friendRequest checking - and good job on unmounting on the bloating components, sound keep the app performance friendly and clean.